lib/server.js

```diff
@@ -813,11 +813,21 @@ Server.prototype._run = function _run(req, res, route, chain, cb) {
     var handlerName = null;
     var errName;
     var emittedError = false;
+    var reqClosed = false;
 
     if (cb) {
         cb = once(cb);
     }
 
+    // attach a listener for 'close' event, this will let us set a flag so that
+    // we can stop processing the request if the client closes the connection
+    // (or we lose the connection).
+    function _requestClose() {
+        reqClosed = true;
```

Comment:
Now that I think about it, we might want to add an additional field to the audit log to indicate that this was a prematurely closed request. Perhaps a field named ```clientClosed```.

Sorry totally my bad for not asking about this earlier.